crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -167,7 +168,11 @@ impl SystemConfigs {
         }
     }
 
-    pub(crate) fn run_if_inner(&mut self, condition: BoxedCondition) {
+    /// Adds a new boxed run condition to the systems.
+    ///
+    /// This is exposed to allow users to create custom scheduling abstractions:
+    /// it is generally only useful when working with boxed run conditions with an unknown dynamic type.
```

Comment:
```suggestion
    /// This is useful if you have a run condition whose concrete type is unknown.
    /// Prefer `run_if` for run conditions whose type is known at compile time.
```

I don't think it makes a lot of sense to talk about the motivation for exposing this function in its docs -- that's a discussion for the PR.